scripts/build.py

```diff
@@ -405,9 +422,11 @@ def process_html(source_file_stream, target_file_stream, file_hashes):
         # This is because html paths are used by backend and we work with
         # paths without hash part in backend.
         if not filepath.endswith('.html'):
+            filepath = common.convert_filepath_to_url(filepath)
```

Comment:
Similar comments as above -- don't rename.

Also, maybe find out what the part of the URL the result corresponds to is actually called -- it's not the whole URL that's being produced, it's just a part of it (right)?

Reply:
Actually it is not a real `url`. Here is the whole picture. The function `process_html` is trying to replace some file path in the compile HTML, the file path in HTML will just be something like `/path/to/a/file` and the `filepath` variable should be exactly the same. While on Windows, the `filepath` variable will be something like `\path\to\a\file` and hence the program will fail to replace the file name. What I want to do here is basically replace the `\` with `/` so this function can successfully replace the path.

Reply:
Thanks for the explanation!

So it seems to me that, conceptually, what you are doing is taking a bunch of filepaths (referencing locations in the local file system) and converting them to URL paths with hashes. You can make a method for that: convert_filepath_to_hashed_url. (That method might use convert_to_posixpath() internally -- see above.)